lib/core/socket-handler.ts

```diff
@@ -34,12 +34,12 @@ export class SocketHandler {
         return handler;
     }
 
-    public constructor (public socket: Socket, private onMessage: (message: string, sequenceNumber?: number) => void) {
+    public constructor (public socket: Socket, private onMessage: (message: string | Buffer, sequenceNumber?: number) => void) {
```

Comment:
I would refactor to create a SocketMessage class (can be inside this file), that:
1) Always stores the buffer as an instance variable
2) Provides a convenience methods for asString(), isString(), and isBinary()
3) Holds the optional sequenceNumber
4) It can also incorporate safe checks like contains(string), which is useful for code like this:
https://github.com/bespoken/bst/blob/c937206df68eff01e6c511a9e83d6be0079ca626/lib/client/bespoke-client.ts#L202

Does that make sense? There is enough going on with the messages now that it makes sense to encapsulate them and add nice safe convenience methods.